### DISCUSSION TITLE
Wave apps and Generator Functions

### DISCUSSION THREAD

**Discussion by optsim** [h2oai/wave#1790] [Q&A]
I am working on a Wave app that control a simulation model, Python package simpy (https://simpy.readthedocs.io/en/latest/index.html). Simpy uses generator functions to advance simulation time, yield statements. After each yield, I update the Wave dashboard with a save. It works with Wave script, but I have trouble to convert it to a Wave app. Are there any successful examples where the page save action takes place in several functions within a class?

**Comment by vopani** [comment]
Some full app examples can be found here:

https://github.com/h2oai/wave-apps
https://github.com/vopani/waveton

**Comment by mturoci** [comment]
Hey @optsim, can you put up a min code example that is not working for you? There should be no difference other than you need to use `await q.page.save()` instead of `page.save`.
